### ISSUE TITLE
Support Android and IOS

### ISSUE THREAD

**Issue by ewhal** [wailsapp/wails#199]
It would be beyond amazing if this was a write once run anywhere type deal.
Supporting mobile devices would really make this happen.
You can run golang on both android and ios.
Unsure about how complicated this would be though.

https://github.com/golang/go/wiki/Mobile
https://godoc.org/golang.org/x/mobile
https://github.com/microo8/gowebview

**Comment by leaanthony** [comment]
Hi! Thanks for opening this request. It has certainly been talked about and we are open to it. This would be something we do post V1 so let's keep this in the backlog.

**Comment by bh90210** [comment]
hi @ewhal 
take my opinion with a grain of salt, as I am far from being an expert on mobile dev (or dev in general lmao).

From my research on the subject I came to the realisation that going with Go for mobile dev is not going to work I'm afraid. 
[Here is a proof-of-concept app I made with gomobile](https://github.com/bh90210/CP500)
This I made using Java + Go. (heck, I even tried writing a [library!](https://github.com/bh90210/SimpleDBW))
What ur looking is actually the third iteration, initially I tried it using [Go Qt bindings](https://github.com/therecipe/qt) and after failure I gave a try with React Native, which again failed me (but if you go the React way it is worth looking at [ghostbridge](https://github.com/ipsn/go-ghostbridge) if u haven't already found it).
_one last thing that seemed fun is [reverse bindings](https://github.com/golang/go/issues/16876)_

The last repository you linked hasn't been update for 3 years. And I take a que from that, gomobile hasn't been updating either. Everything around gomobile is sparse and oldish.

Main three reason I concluded _it will never work on production_ is
a. app size. on mobile small size *is important* having a 30mb minimum size application is subpar in any scenario.
b. if you are writing a library you won't be able to distribute it via maven etc which *is problematic* for adoption.
c. the complexity and time added in re-compiling the go binding it simply doesn't worth the final result. You probably could have done the same thing a bit clearer, easier and faster going native.

bottom line, from my little experience if I was to re-write my app I would avoid React Native/Qt like catch-all solutions and straight go to kotlin (or swift respectively) or at least pure Java (or C#).

**Comment by leaanthony** [comment]
Closing for now and listing on the https://github.com/wailsapp/wails/wiki/Post-V1-Enhancements page

**Comment by LiveLaughLizard** [comment]
@leaanthony Regarding that v1 enhancements link, is the wiki private? I can't view it - it takes me straight to the repo root, and I see no wiki tab. No worries if it isn't open to the public, just curious.
